libpromises/files_names.c

```diff
@@ -589,6 +623,35 @@ bool CompressPath(char *dest, size_t dest_size, const char *src)
 
 /*********************************************************************/
 
+char *GetAbsolutePath(const char *path)
+/**
+ * WARNING: Remember to free return value.
+ **/
+{
+    char abs_path[PATH_MAX] = { 0 };
+    if (IsAbsoluteFileName(path))
+    {
+        CompressPath(abs_path, PATH_MAX, path);
+        return xstrdup(abs_path);
+    }
+    else
+    {
+        /* the full_path can potentially be long (with many '../' parts)*/
+        char full_path[2 * PATH_MAX] = { 0 };
+        if (!getcwd(full_path, PATH_MAX))
```

Comment:
Explicit comparison with NULL pointer. Use the `!` operator only for boolean comparisons.

Reply:
Done.